src/rust/src/asn1_exp/encode.rs

```diff
@@ -0,0 +1,69 @@
+// This file is dual licensed under the terms of the Apache License, Version
+// 2.0, and the BSD License. See the LICENSE file in the root of this repository
+// for complete details.
+
+use asn1::{SimpleAsn1Writable, Writer};
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyString};
+
+use crate::asn1_exp::types::{AnnotatedType, AnnotatedTypeObject, Type};
+
+fn write_value<T: SimpleAsn1Writable>(
```

Comment:
This doesn't seem to do very much?

Reply:
yeah, it's because it's useful to avoid duplication only when all the other types are implemented. See [here](https://github.com/trailofbits/cryptography-asn1/blob/935f7458c23ea8c7ad14e5119a5d71e6fafd67ae/rust/src/encode.rs#L36-L46) (it saves us writing that `match` statement once for each type)